src/editor/lib/entity.js

```diff
@@ -22,20 +22,37 @@ import {
  * @param {string|number|object} value - New value.
  */
 export function updateEntity(entity, component, property, value) {
-  if (property) {
+  // Special handling for vector components (position, rotation, scale)
+  const vectorComponents = ['position', 'rotation', 'scale'];
+
+  if (property && vectorComponents.includes(component)) {
+    // Get current values
+    const currentValues = entity.getAttribute(component) || {};
+
+    if (value === null || value === undefined) {
+      // Remove property
+      delete currentValues[property];
```

Comment:
You will have issue again if you set a partial xyz in aframe.
If you want to use value null or undefined to reset a coordinate, then set it to 0 (position and rotation) or 1 (scale) instead of removing.

Reply:
decided to remove and not handle that case as it's not supposed to happen, better to throw an error
7f4cad1e20c457ec30591dd0d5eea1ae47bd988c